network/f5/bigip_gtm_pool.py

```diff
@@ -0,0 +1,274 @@
+                                    'vs_capacity', 'least_conn', 'lowest_rtt', 'lowest_hops',
+                                    'packet_rate', 'cpu', 'hit_ratio', 'qos', 'bps',
+                                    'drop_packet', 'explicit_ip', 'connection_rate', 'vs_score']
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            server = dict(type='str', required=True),
+            user = dict(type='str', required=True),
+            password = dict(type='str', required=True),
+            state = dict(type='str', required=True, choices=['present', 'absent', 'enabled', 'disabled']),
+            pool = dict(type='str', required=True),
+            partition = dict(type='str', default='Common'),
+            virtual_server_server = dict(type='str', required=False),
+            virtual_server_name = dict(type='str', required=False),
+            lb_method = dict(type='str', required=True, choices=lb_method_choices)
+        ),
+        supports_check_mode=True
+    )
+
+    if not bigsuds_found:
+        module.fail_json(msg="the python bigsuds library is required")
+
+    server = module.params['server']
+    user = module.params['user']
+    password = module.params['password']
+    state = module.params['state']
+    partition = module.params['partition']
+    pool = "/%s/%s" % (partition, module.params['pool'])
+    virtual_server_name = module.params['virtual_server_name']
+    virtual_server_server = module.params['virtual_server_server']
+    lb_method = module.params['lb_method']
+
+    try:
+        api = bigip_api(server, user, password)
+
+        result = {'changed': False}  # default
+
+        if state == 'absent':
+            if pool and virtual_server_name and virtual_server_server:
+                if pool_exists(api, pool) and member_exists(api, pool, virtual_server_name, virtual_server_server):
+                    if not module.check_mode:
+                        remove_pool_member(api, pool, virtual_server_name, virtual_server_server)
+                        remove_pool(api, pool)
+                        result = {'changed': True}
+                    else:
+                        # check-mode return value
+                        result = {'changed': True}
+            elif pool_exists(api, pool):
+                if not module.check_mode:
+                    remove_pool(api, pool)
+                    result = {'changed': True}
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+        elif state == 'present':
+            update = False
+            if not pool_exists(api, pool):
+                if not module.check_mode:
+                    try:
+                        add_pool(api, pool, lb_method)
+                        result = {'changed': True}
+                    except bigsuds.OperationFailed, e:
+                        if "already exists" in str(e):
```

Comment:
I followed the paradigm of the LTM module. I personally haven't seen this race condition. But from the LTM comments:

                    # a bit of a hack to handle concurrent runs of this module.
                    # even though we've checked the pool doesn't exist,
                    # it may exist by the time we run create_pool().
                    # this catches the exception and does something smart
                    # about it!